packages/fx-core/src/component/driver/deploy/azure/impl/azureDeployImpl.ts

```diff
@@ -89,6 +91,27 @@ export abstract class AzureDeployImpl extends BaseDeployImpl {
     azureCredential: TokenCredential
   ): Promise<void>;
 
+  /**
+   * cleanup function after deployment is finished
+   * @protected
+   */
+  protected async cleanup(): Promise<void> {
+    if (this.zipFilePath && !this.dryRun && fs.existsSync(this.zipFilePath)) {
+      try {
+        await fs.remove(this.zipFilePath);
+        // if upper folder is empty, remove it
+        const parentFolder = path.dirname(this.zipFilePath);
+        if ((await fs.readdir(parentFolder)).length === 0) {
```

Comment:
It seems that the `zipFilePath` can be entered by the user. Removing the parent folder may also delete unexpected files. 
For instance, if the user specifies the zip file path as `C:/deployment.zip`, will this code remove all files from disk C?
https://github.com/OfficeDev/TeamsFx/wiki/Available-actions-in-Teams-Toolkit#azurefunctionszipdeploy